src/plugins/workspace/public/components/workspace_updater/workspace_updater.tsx

```diff
@@ -111,8 +114,17 @@ export const WorkspaceUpdater = (props: WorkspaceUpdaterProps) => {
   );
 
   useEffect(() => {
-    setCurrentWorkspaceFormData(getFormDataFromWorkspace(currentWorkspace));
-  }, [currentWorkspace]);
+    const rawFormData = getFormDataFromWorkspace(currentWorkspace);
+
+    if (rawFormData && savedObjects && currentWorkspace) {
+      getDataSourcesList(savedObjects.client, [currentWorkspace.id]).then((selectedDataSources) => {
+        setCurrentWorkspaceFormData({
+          ...rawFormData,
+          selectedDataSources,
+        });
+      });
+    }
```

Comment:
This is the drawback that we add label into the selectedDataSources, you need to fulfill the selected data source id by looking in the data source list. If we only use id as selectedDataSource, then we do not need the getDataSourcesList call here.

Reply:
Do you mean we stored selected data sources ids in workspace attributes? I maybe confused about this if we store a new field in workspace because its value may not be reliable as in savedObject.

Reply:
Oh, my bad. Anyway we need to fetch the list of data source. nvm.